skops/io/_general.py

```diff
@@ -405,6 +407,23 @@ def object_get_state(obj: Any, save_context: SaveContext) -> dict[str, Any]:
     except Exception:
         pass
 
+    # Then we check if the output of __reduce__ is of the form
+    # (constructor, (constructor_args,))
+    # If the constructor is the same as the object's type, then we consider it
+    # safe to call it with the specified arguments.
+
+    reduce_output = obj.__reduce__()
+    # note that we do "=="" to compare types instead of "is", since we only accept
+    # exact matches here.
```

Comment:
I don't understand this comment. Checking `reduce_output[0] == type(obj)` is fine but when would using `is` not be fine too?

Reply:
I somehow thought `is` applies to subclasses, but it doesn't. So changing to `is`